types/jrf-pip/index.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for jrf-pip 1.0
+// Project: https://github.com/jirufik/jrf-pip#readme
+// Definitions by: rufus <https://github.com/jirufik>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface ParallelProcessingParams {
+    arrayValues: any[];
+
+    processingFn(params: ProcessingFnParams): void | Promise<void>;
+
+    nextValueFn?(params: NextValueFnParams): boolean | Promise<boolean>;
+
+    cycleTimeout?: number;
+    parallel?: number;
+    awaitRes?: boolean;
+
+    cb?(stackError: StackError): void | Promise<void>;
+}
+
+interface ProcessingFnParams {
+    value: any;
+    index: number;
+    arrayValues: any[];
+    iteration: number;
+}
+
+interface NextValueFnParams {
+    value: any;
+    index: number;
+    arrayValues: any[];
+    iteration: number;
+}
+
+interface StackError {
+    value: any;
+    index: number;
+    iteration: number;
+    error: any;
+}
+
+declare function parallelProcessing(params: ParallelProcessingParams): Promise<StackError[]>;
```

Comment:
The promise resolves to a `StackError[]` only if `awaitRes` was true. You could model this with an overload, but I don’t know if it’s worth it—I’d just say the return type is `Promise<StackError[] | undefined>`

Reply:
Thanks. Really much better that way.